lib/rule-context.js

```diff
@@ -82,6 +83,9 @@ function RuleContext(ruleId, eslint, severity, options, settings, parserOptions,
     this.parserPath = parserPath;
     this.meta = meta;
 
+    // create a separate copy and freeze it (it's not nice to freeze other people's objects)
+    this.parserServices = Object.freeze(Object.assign({}, parserServices));
```

Comment:
To be clear, I saw `Object.freeze()` at the time of my review and I asked for a *test* around freezing. :smile: If we want to remove `Object.freeze()` and thus not write a test around it, I have no objections. If we are already cloning, maybe we don't need to freeze.